src/EditorFeatures/Core.Wpf/Options/EditorConfigDocumentOptionsProvider.cs

```diff
@@ -59,7 +66,63 @@ private void Workspace_DocumentOpened(object sender, DocumentEventArgs e)
         {
             lock (_gate)
             {
-                _openDocumentContexts.Add(e.Document.Id, Task.Run(() => GetConventionContextAsync(e.Document.FilePath, CancellationToken.None)));
+                var documentId = e.Document.Id;
+                var filePath = e.Document.FilePath;
+                _openDocumentContexts.Add(documentId, Task.Run(async () =>
+                {
+                    var context = await GetConventionContextAsync(filePath, CancellationToken.None).ConfigureAwait(false);
+                    context.CodingConventionsChangedAsync += (innerSender, innerE) => HandleCodingConventionsChangedAsync(documentId, innerSender, innerE);
+                    return context;
+                }));
+            }
+        }
+
+        private static readonly ConditionalWeakTable<CodingConventionsChangedEventArgs, HashSet<ProjectId>> s_projectNotifications =
+            new ConditionalWeakTable<CodingConventionsChangedEventArgs, HashSet<ProjectId>>();
+
+        private Task HandleCodingConventionsChangedAsync(DocumentId documentId, object sender, CodingConventionsChangedEventArgs e)
+        {
+            var projectId = documentId.ProjectId;
+            var projectsAlreadyNotified = s_projectNotifications.GetOrCreateValue(e);
```

Comment:
:memo: This handles aggregation of events by taking advantage of a currently-undocumented condition where any given change to a .editorconfig file on disk maps to a specific instance of `CodingConventionsChangedEventArgs` which is then reused for all context notifications resulting from that change.